src/PackageSelection/PackageSelection.php

```diff
@@ -186,6 +193,27 @@ public function select(Composer $composer, $verbose)
 
         $links = $this->requireAll ? $this->getAllLinks($repos, $this->minimumStability, $verbose) : $this->getFilteredLinks($composer);
 
+        if ($this->bestCandidateStrategy) {
+            // split the links with OR in order to manage them alone
+            foreach ($links as $key => $link) {
+                $constraint = $link->getConstraint();
+                if ($constraint instanceof MultiConstraint && !$constraint->isConjunctive()) {
+                    foreach ($constraint->getConstraints() as $newConstraint) {
+                        $links[] = new Link(
+                            $link->getSource(),
+                            $link->getTarget(),
+                            $newConstraint,
+                            $link->getDescription(),
+                            $newConstraint->getPrettyString()
+                        );
+                    }
+                    unset($links[$key]);
+                }
+            }
+            /** @var Link[] $links */
+            $links = array_merge($links);
```

Comment:
merging a single array ? Why ? If you want to re-index it, use ``array_values`` instead. It would be more more understandable